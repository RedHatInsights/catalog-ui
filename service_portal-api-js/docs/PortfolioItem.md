# ServicePortalApi.PortfolioItem

## Properties
Name | Type | Description | Notes
------------ | ------------- | ------------- | -------------
**id** | **String** |  | [optional] 
**favorite** | **Boolean** |  | [optional] 
**name** | **String** |  | [optional] 
**description** | **String** |  | [optional] 
**orphan** | **Boolean** |  | [optional] 
**state** | **String** |  | [optional] 


