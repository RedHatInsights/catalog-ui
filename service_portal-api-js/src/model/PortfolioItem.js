/**
 * Service Portal API
 * This is a API to fetch and order catalog items from different cloud sources
 *
 * OpenAPI spec version: 1.0.0
 * Contact: you@your-company.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 *
 */


import ApiClient from '../ApiClient';





/**
* The PortfolioItem model module.
* @module model/PortfolioItem
* @version 1.0.0
*/
export default class PortfolioItem {
    /**
    * Constructs a new <code>PortfolioItem</code>.
    * @alias module:model/PortfolioItem
    * @class
    */

    constructor() {
        

        
        

        

        
    }

    /**
    * Constructs a <code>PortfolioItem</code> from a plain JavaScript object, optionally creating a new instance.
    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
    * @param {Object} data The plain JavaScript object bearing properties of interest.
    * @param {module:model/PortfolioItem} obj Optional instance to populate.
    * @return {module:model/PortfolioItem} The populated <code>PortfolioItem</code> instance.
    */
    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new PortfolioItem();

            
            
            

            if (data.hasOwnProperty('id')) {
                obj['id'] = ApiClient.convertToType(data['id'], 'String');
            }
            if (data.hasOwnProperty('favorite')) {
                obj['favorite'] = ApiClient.convertToType(data['favorite'], 'Boolean');
            }
            if (data.hasOwnProperty('name')) {
                obj['name'] = ApiClient.convertToType(data['name'], 'String');
            }
            if (data.hasOwnProperty('description')) {
                obj['description'] = ApiClient.convertToType(data['description'], 'String');
            }
            if (data.hasOwnProperty('orphan')) {
                obj['orphan'] = ApiClient.convertToType(data['orphan'], 'Boolean');
            }
            if (data.hasOwnProperty('state')) {
                obj['state'] = ApiClient.convertToType(data['state'], 'String');
            }
        }
        return obj;
    }

    /**
    * @member {String} id
    */
    id = undefined;
    /**
    * @member {Boolean} favorite
    */
    favorite = undefined;
    /**
    * @member {String} name
    */
    name = undefined;
    /**
    * @member {String} description
    */
    description = undefined;
    /**
    * @member {Boolean} orphan
    */
    orphan = undefined;
    /**
    * @member {String} state
    */
    state = undefined;








}


