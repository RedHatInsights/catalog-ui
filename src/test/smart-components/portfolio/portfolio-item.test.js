import React from 'react';
import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import promiseMiddleware from 'redux-promise-middleware';
import { notificationsMiddleware } from '@redhat-cloud-services/frontend-components-notifications/';

import PortfolioItem from '../../../smart-components/portfolio/portfolio-item';

describe('<PortfolioItem />', () => {
  let initialProps;
  let initialState;
  const middlewares = [thunk, promiseMiddleware, notificationsMiddleware()];
  let mockStore;

  const ComponentWrapper = ({ store, children }) => (
    <Provider store={store}>
      <MemoryRouter>{children}</MemoryRouter>
    </Provider>
  );

  beforeEach(() => {
    initialProps = {
      orderUrl: '/order',
      id: '1',
      name: 'Foo',
      description: 'Bar',
      display_name: 'quux',
      metadata: {
        user_capabilities: {
          destroy: true
        }
      }
    };
    initialState = {
      platformReducer: {
        platformIconMapping: {}
      },
      portfolioReducer: {
        portfolioItems: {
          data: [
            {
              orderUrl: '/order',
              id: '1',
              name: 'Foo',
              description: 'Bar',
              display_name: 'quux',
              metadata: {
                user_capabilities: {
                  destroy: true
                }
              }
            }
          ]
        }
      }
    };
    mockStore = configureStore(middlewares);
  });

  it('should render correctly', (done) => {
    const store = mockStore(initialState);
    const wrapper = mount(
      <ComponentWrapper store={store}>
        <PortfolioItem {...initialProps} />
      </ComponentWrapper>
    );
    setImmediate(() => {
      expect(shallowToJson(wrapper.find(PortfolioItem))).toMatchSnapshot();
      done();
    });
  });

  it('should check the item correctly', (done) => {
    const onSelect = jest.fn();
    const store = mockStore(initialState);
    const wrapper = mount(
      <ComponentWrapper store={store}>
        <PortfolioItem {...initialProps} onSelect={onSelect} isSelectable />
      </ComponentWrapper>
    );
    setImmediate(() => {
      wrapper
        .find(PortfolioItem)
        .find('input')
        .simulate('change');
      expect(onSelect).toHaveBeenCalledWith('1');
      done();
    });
  });

  it('should not render checkbox if capability destroy is set to false', () => {
    const onSelect = jest.fn();
    const store = mockStore(initialState);
    const wrapper = mount(
      <ComponentWrapper store={store}>
        <PortfolioItem
          {...initialProps}
          metadata={{
            user_capabilities: {
              destroy: false
            }
          }}
          onSelect={onSelect}
          isSelectable
        />
      </ComponentWrapper>
    );
    expect(wrapper.find(PortfolioItem).find('input')).toHaveLength(0);
  });
});
