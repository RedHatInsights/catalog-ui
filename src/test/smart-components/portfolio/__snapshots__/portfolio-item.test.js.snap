// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PortfolioItem /> should render correctly 1`] = `
<PortfolioItem
  description="Bar"
  display_name="quux"
  id="1"
  metadata={
    Object {
      "user_capabilities": Object {
        "destroy": true,
      },
    }
  }
  name="Foo"
  orderUrl="/order"
>
  <Styled(Component)>
    <Component
      className="sc-fznyAO hahLsc"
    >
      <Component
        className="sc-fznyAO hahLsc"
      >
        <div
          className="sc-fznyAO hahLsc"
        >
          <Styled(Component)>
            <Component
              className="sc-fzplWN kZIWSH"
            >
              <article
                className="pf-c-card sc-fzplWN kZIWSH"
              >
                <Component>
                  <div
                    className="pf-c-card__header"
                  >
                    <Styled(Component)>
                      <Component
                        className="sc-fznKkj jHfxlR"
                      >
                        <div
                          className="pf-l-level sc-fznKkj jHfxlR"
                        >
                          <CardIcon
                            height={40}
                            src="/api/catalog/v1.1/portfolio_items/1/icon"
                            style={Object {}}
                          >
                            <styled.div
                              height={40}
                            >
                              <div
                                className="sc-AxheI eDjZka"
                                height={40}
                              >
                                <IconPlaceholder
                                  height={40}
                                >
                                  <svg
                                    height={40}
                                    width={40}
                                  >
                                    <circle
                                      cx={20}
                                      cy={20}
                                      fill="#ecebeb"
                                      r={20}
                                    />
                                  </svg>
                                </IconPlaceholder>
                                <Styled(t)
                                  delayTime={0}
                                  height={0}
                                  hidden={true}
                                  onError={[Function]}
                                  onLoad={[Function]}
                                  src="/api/catalog/v1.1/portfolio_items/1/icon"
                                  threshold={2000}
                                >
                                  <t
                                    afterLoad={[Function]}
                                    beforeLoad={[Function]}
                                    className="sc-AxgMl hkMtbB"
                                    delayMethod="throttle"
                                    delayTime={0}
                                    effect=""
                                    height={0}
                                    hidden={true}
                                    onError={[Function]}
                                    onLoad={[Function]}
                                    placeholderSrc={null}
                                    src="/api/catalog/v1.1/portfolio_items/1/icon"
                                    threshold={2000}
                                    useIntersectionObserver={true}
                                    visibleByDefault={false}
                                    wrapperClassName=""
                                  >
                                    <t
                                      afterLoad={[Function]}
                                      beforeLoad={[Function]}
                                      className="sc-AxgMl hkMtbB"
                                      delayMethod="throttle"
                                      delayTime={0}
                                      height={0}
                                      threshold={2000}
                                      useIntersectionObserver={true}
                                      visibleByDefault={false}
                                    >
                                      <img
                                        className="sc-AxgMl hkMtbB"
                                        height={0}
                                        hidden={true}
                                        onError={[Function]}
                                        onLoad={[Function]}
                                        src="/api/catalog/v1.1/portfolio_items/1/icon"
                                      />
                                    </t>
                                  </t>
                                </Styled(t)>
                              </div>
                            </styled.div>
                          </CardIcon>
                        </div>
                      </Component>
                    </Styled(Component)>
                  </div>
                </Component>
                <ServiceOfferingCardBody
                  description="Bar"
                  display_name="quux"
                  id="1"
                  metadata={
                    Object {
                      "user_capabilities": Object {
                        "destroy": true,
                      },
                    }
                  }
                  name="Foo"
                  orderUrl="/order"
                >
                  <Styled(Component)>
                    <Component
                      className="sc-fzpans hEOQak"
                    >
                      <div
                        className="pf-c-card__body sc-fzpans hEOQak"
                      >
                        <Component>
                          <div
                            className="pf-c-content"
                          >
                            <ConditionalLink>
                              <Component
                                component="h3"
                                title="Foo"
                              >
                                <h3
                                  className=""
                                  data-pf-content={true}
                                  title="Foo"
                                >
                                  <styled.div>
                                    <div
                                      className="sc-fzoLsD VNciv"
                                    >
                                      Foo
                                    </div>
                                  </styled.div>
                                </h3>
                              </Component>
                            </ConditionalLink>
                          </div>
                        </Component>
                        <ItemDetails
                          description="Bar"
                          id="1"
                          metadata={
                            Object {
                              "user_capabilities": Object {
                                "destroy": true,
                              },
                            }
                          }
                          orderUrl="/order"
                          toDisplay={
                            Array [
                              "description",
                            ]
                          }
                        >
                          <styled.div
                            key="card-prop-description"
                          >
                            <div
                              className="sc-Axmtr bQyRiD"
                            >
                              Bar
                            </div>
                          </styled.div>
                        </ItemDetails>
                      </div>
                    </Component>
                  </Styled(Component)>
                </ServiceOfferingCardBody>
                <Component>
                  <div
                    className="pf-c-card__footer"
                  />
                </Component>
              </article>
            </Component>
          </Styled(Component)>
        </div>
      </Component>
    </Component>
  </Styled(Component)>
</PortfolioItem>
`;
